apps/block_scout_web/lib/block_scout_web/web_router.ex

```diff
@@ -78,7 +79,13 @@ defmodule BlockScoutWeb.WebRouter do
 
     resources("/bridged-tokens", BridgedTokensController, only: [:index, :show])
 
-    resources "/address", AddressController, only: [:show] do
+    address_validation_options =
```

Comment:
I would perhaps remove this, it should work fine without it now when you added the check.

Reply:
Because this PR optionally validates parameters at the Plug level, it's not possible to pass invalid parameters to the Controllers under test and assert their validation logic without disabling it.

Controllers test this by [running a `Plug.Conn` object through the Router](https://github.com/celo-org/blockscout/blob/02a6c100aadc70f235f9d71a85caddae31586a3c/apps/block_scout_web/test/block_scout_web/controllers/address_read_proxy_controller_test.exs#L25), which includes running through [plug pipelines defined for this route](https://github.com/celo-org/blockscout/blob/1c1357f40f9c59d0c69098c40867cd30130c24f8/apps/block_scout_web/lib/block_scout_web/web_router.ex#L24). When this is done all the Plug functions are invoked and the Conn object is manipulated appropriately.

Put simply, this PR adds the ability to filter out invalid parameters before the Controllers receive them. I'm not so happy with adding this conditional in the router, but the alternative would be to remove validation tests for each controller that asserts the format of an address. Given that the behaviour of the app has changed here that would make sense, the tests now assert behaviour that represents an invalid state of the application. On the other hand I feel rather uneasy removing these tests, given that they are assert something in it's concrete domain and the new solution is an instance of a generic implementation that is incidentally configured to align with the domain (in this case, address hash format).

Remove controller tests + remove conditional, or keep awkward conditional in the router - what do you think?

Reply:
That fact that we're adding the new validation behavior would change the way system reacts to invalid address in real life, right? Meaning, that the behavior that those tests are checking won't be observed any more. The new behavior is covered by the test you've added in this PR.
So, if we're adding this check here we're leaving "dead" tests in the codebase just for the sake of testing that this check works. I think removing those tests should be okay, unless I'm missing something obvious.